src/overlays/actors/ovl_En_Nb/z_en_nb.c

```diff
@@ -1,15 +1,1553 @@
+    } else {
+        unk0 = frameCount;
+        fc = 0.0f;
+        playbackSpeed = -1.0f;
+    }
+
+    SkelAnime_ChangeAnim(&this->skelAnime, animation, playbackSpeed, unk0, fc, mode, transitionRate);
+}
+
+void EnNb_SetChamberAnim(EnNb* this, GlobalContext* globalCtx) {
+    EnNb_SetCurrentAnim(this, &D_06009694, 0, 0, 0);
+    this->actor.shape.unk_08 = -10000.0f;
+}
+
+void EnNb_SpawnBlueWarp(EnNb* this, GlobalContext* globalCtx) {
+    f32 posX = this->actor.posRot.pos.x;
+    f32 posY = this->actor.posRot.pos.y;
+    f32 posZ = this->actor.posRot.pos.z;
+
+    Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_DOOR_WARP1, posX, posY, posZ, 0, 0, 0, 2);
+}
+
+void EnNb_GiveMedallion(EnNb* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    f32 posX = player->actor.posRot.pos.x;
+    f32 posY = player->actor.posRot.pos.y + 50.0f;
+    f32 posZ = player->actor.posRot.pos.z;
+
+    Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_DEMO_EFFECT, posX, posY, posZ, 0, 0, 0,
+                       0xC);
+    Item_Give(globalCtx, ITEM_MEDALLION_SPIRIT);
+}
+
+void EnNb_ComeUpImpl(EnNb* this, GlobalContext* globalCtx) {
+    this->actor.shape.unk_08 += 83.333336f;
+}
+
+void EnNb_SetupChamberCsImpl(EnNb* this, GlobalContext* globalCtx) {
+    s32 pad[2];
+    Player* player;
+    s16 yaw;
+
+    if ((gSaveContext.chamberCutsceneNum == 3) && (gSaveContext.sceneSetupIndex < 4)) {
+        player = PLAYER;
+        this->action = NB_CHAMBER_UNDERGROUND;
+        globalCtx->csCtx.segment = &D_80AB431C;
+        gSaveContext.cutsceneTrigger = 2;
+        Item_Give(globalCtx, ITEM_MEDALLION_SPIRIT);
+        yaw = this->actor.posRot.rot.y + 0x8000;
+        player->actor.shape.rot.y = yaw;
+        player->actor.posRot.rot.y = yaw;
+    }
+}
+
+void EnNb_SetupChamberWarpImpl(EnNb* this, GlobalContext* globalCtx) {
+    CutsceneContext* csCtx = &globalCtx->csCtx;
+    CsCmdActorAction* csCmdNPCAction;
+
+    if (csCtx->state != 0) {
+        csCmdNPCAction = csCtx->npcActions[1];
+        if (csCmdNPCAction != NULL && csCmdNPCAction->action == 2) {
+            this->action = NB_CHAMBER_APPEAR;
+            this->drawMode = NB_DRAW_DEFAULT;
+            EnNb_SpawnBlueWarp(this, globalCtx);
+        }
+    }
+}
+
+void EnNb_SetupDefaultChamberIdle(EnNb* this) {
+    if (this->actor.shape.unk_08 >= 0.0f) {
+        this->action = NB_CHAMBER_IDLE;
+        this->actor.shape.unk_08 = 0.0f;
+    }
+}
+
+void EnNb_SetupArmRaise(EnNb* this, GlobalContext* globalCtx) {
+    AnimationHeader* animation = &D_0600274C;
+    CsCmdActorAction* csCmdNPCAction;
+
+    if (globalCtx->csCtx.state != 0) {
+        csCmdNPCAction = globalCtx->csCtx.npcActions[1];
+        if (csCmdNPCAction != NULL && csCmdNPCAction->action == 3) {
+            SkelAnime_ChangeAnim(&this->skelAnime, animation, 1.0f, 0.0f,
+                                 SkelAnime_GetFrameCount(&animation->genericHeader), 2, 0.0f);
```

Comment:
Might be worth going through and removing any genericHeader access in this file, unless we plan to keep that for a future skelanime cleanup.

Reply:
i believe petrie already cleaned that up in his fixes, so ill go ahead and do that here